src/server/routes/annualData.js

```diff
@@ -1,62 +1,195 @@
+import modelListOptions from '../../app/utils/ModelListOptionsService';
+import { matchListDate } from '../../app/utils/DateService';
 
 const nyplApiClientGet = endpoint =>
   nyplApiClient().then(client => client.get(endpoint, { cache: false }));
 
-/* annualCurrentData
+/**
+ * annualCurrentListData(req, res, next)
  * Get the latest annual staff pick list for either childrens or ya.
+ * @param {object} req
+ * @param {object} res
+ * @param {object} next
  */
-function annualCurrentData(type, req, res, next) {
-  const pageTitle = appConfig.pageTitle[type];
-  const metaTags = appConfig.metaTags[type];
-  let dataType = '';
-
-  if (type === 'childrens') {
-    dataType = 'kids';
-  } else if (type === 'ya') {
-    dataType = 'teens';
-  }
+function annualCurrentListData(req, res, next) {
+  const listOptions = config.annualListOptions;
+  let seasonListOptions = [];
+  let latestSeason = '';
+  const dataType = utils.getDataType(req.params.type);
 
-  nyplApiClientGet(`/book-lists/${dataType}/2017`)
+  nyplApiClientGet(platformConfig.endpoints.annualLists[dataType])
+    .then((data) => {
+      // Models the options based on the data returned
+      const modeledOptionObject = modelListOptions(data, req.params.type);
+
+      seasonListOptions = modeledOptionObject.options;
+      latestSeason = modeledOptionObject.latestOption;
+
+      // Updates default season list options with API response
+      listOptions.season.options = seasonListOptions;
+
+      // Calls the latest list
+      return nyplApiClientGet(`${platformConfig.endpoints.annualPath}${dataType}/${latestSeason}`);
+    })
     .then((data) => {
       const filters = utils.getAllTags(data.picks);
       // Get the subset of tags that the picks can be filtered by.
       const selectableFilters = utils.getSelectableTags(data.picks);
 
+      // If error returned from the endpoint
+      if (data.statusCode >= 400) {
+        console.error(`Status Code: ${data.statusCode}, Error Message: ${data.error}`);
+
+        return res.redirect(`${config.baseUrl}404`);
+      }
+
+      // Uodate the option lists' default values by the request params
+      listOptions.season.currentValue = latestSeason;
+
       res.locals.data = {
         BookStore: {
           filters,
           picksData: data,
           selectableFilters,
           isJsEnabled: false,
+          listOptions,
+          currentSeason: latestSeason,
         },
-        pageTitle,
-        metaTags,
+        pageTitle: '',
+        metaTags: [],
       };
 
       next();
     })
     .catch((error) => {
-      console.log(`Error fetching endpoint: ${error}`);
+      console.error(`Status Code: ${error.statusCode}, Error Message: ${error.code}`);
+
+      return res.redirect(`${config.baseUrl}404`);
+    });
+}
+
+/**
+ * annualListData(req, res, next)
+ * Get a specific month's or season's staff pick list.
+ * It calls '/book-lists?type=staff-picks' to get all the available list options first.
```

Comment:
Some of the comments and variables should be updated to reflect the annual descriptions.